smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelValidator.java

```diff
@@ -64,307 +80,302 @@ private static final class LazyValidatorFactoryHolder {
-                .build();
-    }
+        private boolean filterPrelude(ValidationEvent event) {
+            // Don't emit any non-error events for prelude shapes and traits.
+            // This prevents custom validators from unnecessarily needing to worry about prelude shapes and trait
+            // definitions, but still allows for validation events when the prelude is broken.
+            return event.getSeverity() == Severity.ERROR || !event.getShapeId()
+                    .filter(Prelude::isPreludeShape)
+                    .isPresent();
+        }
 
-    private static void loadModelSuppressions(
-            List<Suppression> suppressions,
-            Model model,
-            List<ValidationEvent> events
-    ) {
-        model.getMetadataProperty(SUPPRESSIONS).ifPresent(value -> {
-            List<ObjectNode> values = value.expectArrayNode().getElementsAs(ObjectNode.class);
-            for (ObjectNode rule : values) {
-                try {
-                    suppressions.add(Suppression.fromMetadata(rule));
-                } catch (SourceException e) {
-                    events.add(ValidationEvent.fromSourceException(e));
+        private ValidationEvent modifyEventSeverity(ValidationEvent event) {
+            // Use a suppress trait if present.
+            if (event.getShapeId().isPresent()) {
+                ShapeId target = event.getShapeId().get();
+                Shape shape = model.getShape(target).orElse(null);
+                if (shape != null) {
+                    if (shape.hasTrait(SuppressTrait.class)) {
+                        Suppression suppression = Suppression.fromSuppressTrait(shape);
+                        if (suppression.test(event)) {
+                            return changeSeverity(event, Severity.SUPPRESSED, suppression.getReason().orElse(null));
+                        }
+                    }
                 }
             }
-        });
-    }
-
-    private static List<ValidationEvent> suppressEvents(
-            Model model,
-            List<ValidationEvent> events,
-            List<Suppression> suppressions) {
-        return events.stream().map(event -> suppressEvent(model, event, suppressions)).collect(Collectors.toList());
-    }
 
-    private static ValidationEvent suppressEvent(Model model, ValidationEvent event, List<Suppression> suppressions) {
-        // ERROR and SUPPRESSED events cannot be suppressed.
-        if (!event.getSeverity().canSuppress()) {
-            return event;
-        }
+            // Check metadata and manual suppressions.
+            for (Suppression suppression : suppressions) {
+                if (suppression.test(event)) {
+                    return changeSeverity(event, Severity.SUPPRESSED, suppression.getReason().orElse(null));
+                }
+            }
 
-        Suppression matchedSuppression = findMatchingSuppression(model, event, suppressions);
+            Severity appliedSeverity = event.getSeverity();
```

Comment:
It looks like the suppression event modifications return immediately, and we allow suppression of WARNING events, but there are also few/no test cases that enforce how suppressions work with this.

Reply:
suppressions take precedence. Will add a test.